codec-base/src/main/java/io/netty/handler/codec/ByteToMessageDecoder.java

```diff
@@ -279,7 +283,18 @@ public final void handlerRemoved(ChannelHandlerContext ctx) throws Exception {
     protected void handlerRemoved0(ChannelHandlerContext ctx) throws Exception { }
 
     @Override
-    public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception {
+    public void channelRead(ChannelHandlerContext ctx, Object input) throws Exception {
+        if (decodeState == STATE_INIT) {
+            do {
+                channelReadOne(ctx, input);
+            } while (!inputMessages.isEmpty() && (input = inputMessages.pollFirst()) != null);
```

Comment:
Just remove the `isEmpty()` check... there is no real benefit imho.
```suggestion
            } while ((input = inputMessages.pollFirst()) != null);
```